src/ctia/lib/riemann.clj

```diff
@@ -0,0 +1,124 @@
+(ns ctia.lib.riemann
+  (:require [clojure.string :as str]
+            [clojure.tools.logging :as log]
+            [riemann.client :as riemann]
+            [ctia.properties :as prop]
+            [ctia.lib.utils :as utils])
+  (:import [clojure.lang ExceptionInfo]))
+
+(defn request->event
+  [request extra-fields]
+  (into {:uri (str (:uri request))
+         :_params (utils/safe-pprint (:params request))
+         :remote-addr (str (if-let [xff (get-in request [:headers "x-forwarded-for"])]
+                             (peek (str/split xff #"\s*,\s*"))
+                             (:remote-addr request)))
+         :request-headers (prn-str (:headers request))
+         :request-body (let [;; HttpInputOverHTTP => string
+                             bstr (pr-str (:body request))]
+                         (subs bstr 0 (min (count bstr) 100)))
+         :request-method (str (:request-method request))
+         :mask (str (= "Mask" (get-in request [:headers "x-client-app"])))
+         :identity (:identity request)
+         :jwt (:jwt request)}
+        extra-fields))
+
+(defn ms-elapsed
+  "provide how much ms were elapsed since `nano-start`."
+  [nano-start]
+  (/ (- (System/nanoTime) nano-start) 1000000.0))
+
+(defn- send-request-metrics [send-event-fn request extra-fields]
+  (let [event (request->event request extra-fields)]
+    (try
+      (send-event-fn event)
+      (catch Exception e
+        (log/warnf "A Problem occured while sending request metrics event:\n\n%s\n\nException:\n%s"
+                   (utils/safe-pprint event)
+                   (utils/safe-pprint e))
+        (when (nil? send-event-fn)
+          (log/warn "The send-event-fn looks empty. This is certainly due to a configuration problem or perhaps simply a code bug."))))))
+
+(defn wrap-request-metrics [handler service-name send-event-fn]
+  (fn [request]
+    (let [start (System/nanoTime)]
+      (try
+        (when-let [response (handler request)]
+          (let [ms (ms-elapsed start)]
+            (send-request-metrics send-event-fn request
+                                  {:metric ms
+                                   :tags ["ctia" "http"]
+                                   :description (str "Response took "
+                                                     (.format (java.text.DecimalFormat. "#.##")
+                                                              (/ ms 1000))
+                                                     " seconds")
```

Comment:
I wouldn't lose resources for repeating that message as a string. I think having a `metric` is clear enough. We try to enforce the use of `ms`.